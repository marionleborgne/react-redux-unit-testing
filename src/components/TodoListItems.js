import React from 'react';
import styled from 'styled-components';

export const TodoListItem = styled.div`
  display: flex;
  justify-content: space-between;
  border-radius: 5px;
  border: 1px solid white;
  padding: 10px;
  margin: 5px 0;
  background: white;
  color: black;
`;

export const ItemText = styled.div`
  text-decoration: ${props => props.done ? 'line-through' : null}
`;
ItemText.displayName = 'ItemText';


export const ItemLink = styled.a`
  cursor: pointer;
  font-style: italic;
`;
ItemLink.displayName = 'ItemLink';

const TodoListItems = props => (
  <div id="todo-list-items">
    {
      props.items.map(item =>
        <TodoListItem key={item.id}>
          <ItemText done={item.done}>
            {item.text}
          </ItemText>
          <ItemLink onClick={() => props.onItemClick(item)}>
            {item.done ? 'Clear' : 'Done'}
          </ItemLink>
        </TodoListItem>
      )
    }
  </div>
);

export default TodoListItems;
