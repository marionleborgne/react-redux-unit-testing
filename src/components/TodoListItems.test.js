import React from 'react';
import { shallow } from 'enzyme';

import TodoListItems, { TodoListItem, ItemText } from './TodoListItems';

const _noop = () => {};

const _createItems = () => [
  { id: 1, text: 'one', done: false },
  { id: 2, text: 'two', done: false },
  { id: 3, text: 'three', done: false }
];

const _createWrapper = items => shallow(
  <TodoListItems
    onItemClick={_noop}
    items={items}
  />
);

describe('TodoListItems shallow tests', () => {

  it('renders', () => {
    const items = _createItems();
    const wrapper = _createWrapper(items);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders todo list items', () => {

    // Expect the wrapper to be defined and to have
    // the right number of list items.

    const items = _createItems();
    const wrapper = _createWrapper(items);

    // https://airbnb.io/enzyme/docs/api/ReactWrapper/find.html
    // The method find() allows you to find elements by passing an EnzymeSelector
    // EnzymeSelectors can be:

    // 1. CSS selectors.
    expect(wrapper.find('#todo-list-items')).toBeDefined();


    // 2. Component Constructors.
    expect(wrapper.find(TodoListItem)).toHaveLength(items.length);

    // 3. Display Name.
    // Note: for instances of styled-component, you need to
    // explicitly set it. E.g: MyStyledComponent.displayName='MyStyled'.
    expect(wrapper.find('ItemText')).toHaveLength(items.length);
  });

  it('renders a list item', () => {
    const items = _createItems();
    const wrapper = _createWrapper(items);

    // Check if an element in the Component exists.
    expect(wrapper.contains(
      <ItemText done={items[0].done}>
        {items[0].text}
      </ItemText>
    )).toBeTruthy();
  });

  it('renders correct text in item', () => {
    const items = _createItems();
    const wrapper = _createWrapper(items);

    //Expect the child of the first item to be an array
    expect(wrapper.find('ItemText').get(0).props.children).toEqual(items[0].text);
  });
});
