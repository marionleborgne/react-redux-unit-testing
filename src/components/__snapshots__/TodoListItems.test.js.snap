// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoListItems tests renders 1`] = `
<div
  id="todo-list-items"
>
  <styled.div
    key="1"
  >
    <ItemText
      done={false}
    >
      one
    </ItemText>
    <ItemLink
      onClick={[Function]}
    >
      Done
    </ItemLink>
  </styled.div>
  <styled.div
    key="2"
  >
    <ItemText
      done={false}
    >
      two
    </ItemText>
    <ItemLink
      onClick={[Function]}
    >
      Done
    </ItemLink>
  </styled.div>
  <styled.div
    key="3"
  >
    <ItemText
      done={false}
    >
      three
    </ItemText>
    <ItemLink
      onClick={[Function]}
    >
      Done
    </ItemLink>
  </styled.div>
</div>
`;
